### ISSUE TITLE
jinja2.exceptions.TemplateNotFound: android\edx.properties

### ISSUE THREAD

**Issue by danielmgzzg** [overhangio/tutor#234]
Hello, I am trying to get this going as I really like the idea and would like to test it out properly. I have a PC computer (main) and a mac. I managed to get pass this error on the mac, however in windows I am getting the following.

The error indicates that templates are not found after running **tutor config save**.

I changed line 14 in env.py as I tought it was an issue with 
from 
TEMPLATES_ROOT = os.path.join(os.path.dirname(__file__), "templates")
to
TEMPLATES_ROOT = os.path.join(os.getcwd(), "templates")

but it's still not working
```
PS C:\Users\Daniel\tutor\tutor> tutor config save
Configuration saved to C:\Users\Daniel\AppData\Local\tutor\tutor\config.yml
Error loading template android\edx.properties
Traceback (most recent call last):
  File "C:\Users\Daniel\Anaconda3\Scripts\tutor-script.py", line 11, in <module>
    load_entry_point('tutor-openedx', 'console_scripts', 'tutor')()
  File "c:\users\daniel\tutor\tutor\commands\cli.py", line 23, in main
    cli()
  File "C:\Users\Daniel\AppData\Roaming\Python\Python37\site-packages\click\core.py", line 764, in __call__
    return self.main(*args, **kwargs)
  File "C:\Users\Daniel\AppData\Roaming\Python\Python37\site-packages\click\core.py", line 717, in main
    rv = self.invoke(ctx)
  File "C:\Users\Daniel\AppData\Roaming\Python\Python37\site-packages\click\core.py", line 1137, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "C:\Users\Daniel\AppData\Roaming\Python\Python37\site-packages\click\core.py", line 1137, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "C:\Users\Daniel\AppData\Roaming\Python\Python37\site-packages\click\core.py", line 956, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "C:\Users\Daniel\AppData\Roaming\Python\Python37\site-packages\click\core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "c:\users\daniel\tutor\tutor\commands\config.py", line 46, in save
    env.save(root, config)
  File "c:\users\daniel\tutor\tutor\env.py", line 98, in save
    render_full(root, config)
  File "c:\users\daniel\tutor\tutor\env.py", line 107, in render_full
    save_subdir(subdir, root, config)
  File "c:\users\daniel\tutor\tutor\env.py", line 133, in save_subdir
    save_file(path, root, config)
  File "c:\users\daniel\tutor\tutor\env.py", line 141, in save_file
    rendered = render_file(config, path)
  File "c:\users\daniel\tutor\tutor\env.py", line 155, in render_file
    return Renderer.render_file(config, os.path.join(*path))
  File "c:\users\daniel\tutor\tutor\env.py", line 53, in render_file
    template = cls.environment(config).get_template(path)
  File "C:\Users\Daniel\Anaconda3\lib\site-packages\jinja2\environment.py", line 830, in get_template
    return self._load_template(name, self.make_globals(globals))
  File "C:\Users\Daniel\Anaconda3\lib\site-packages\jinja2\environment.py", line 804, in _load_template
    template = self.loader.load(self, name, globals)
  File "C:\Users\Daniel\Anaconda3\lib\site-packages\jinja2\loaders.py", line 113, in load
    source, filename, uptodate = self.get_source(environment, name)
  File "C:\Users\Daniel\Anaconda3\lib\site-packages\jinja2\loaders.py", line 168, in get_source
    pieces = split_template_path(template)
  File "C:\Users\Daniel\Anaconda3\lib\site-packages\jinja2\loaders.py", line 31, in split_template_path
    raise TemplateNotFound(template)
jinja2.exceptions.TemplateNotFound: android\edx.properties

```
**Environment:**
Windows 10
Tutor version 3.5.3

**Additional context**
Anaconda 1.7.2 (Python 3.7.3)

Thank you.

**Comment by regisb** [comment]
Hi @danielmgzzg! Tutor should work in the Ubuntu console for Windows, but it's a bold and ambitious endeavour to try to run it on DOS. (Wait, is the Windows command line still called DOS these days?) Unfortunately, I don't have the resources to test Tutor on Windows. I suggest you run Tutor on your Mac or on a VPS (virtual private server).

Is it ok if I close this?

**Comment by danielmgzzg** [comment]
ok, I get it. Will try it in ubuntu console but maybe I'll just use a Mac.

Thank you

Best,

Daniel

**Comment by mike-holcomb** [comment]
The issue may be that jinja2 always uses `/` for its resource location separators [regardless of the file system](https://github.com/pallets/jinja/issues/411), but `tutor/env.py#walk_templates(subdir, root)` uses filesystem path separators.  This ugly hack at least gets me past the config steps:
```
def walk_templates(subdir, root=TEMPLATES_ROOT):
    """
    Iterate on the template files from `templates/<subdir>`.

    Yield:
        path: template path relative to the template root
    """
    for dirpath, _, filenames in os.walk(template_path(subdir)):
        if not is_part_of_env(dirpath):
            continue
        for filename in filenames:
            path = os.path.join(os.path.relpath(dirpath, root), filename)
+           path = path.replace('\\','/')
            if is_part_of_env(path):
                yield path
```

**Comment by regisb** [comment]
@mike-holcomb Thanks for your input. I'm not ready to provide Windows support for Tutor, but if there is an easy hack that makes it possible to run tutor on Windows then I'll be glad to oblige. Let me re-open this and I'll fix this for the next release.

**Comment by regisb** [comment]
@danielmgzzg @mike-holcomb can you please confirm this is fixed in the latest tutor release? (v3.11.3)
